src/client/Microsoft.Identity.Client/ManagedIdentityApplication.cs

```diff
@@ -45,6 +45,13 @@ public sealed class ManagedIdentityApplication
         /// <inheritdoc/>
         public AcquireTokenForManagedIdentityParameterBuilder AcquireTokenForManagedIdentity(string resource)
         {
+            if (string.IsNullOrEmpty(resource))
+            {
+                throw new MsalClientException(
```

Comment:
Throw ArgumentNullException instead. We shouldn't invent exceptions for this.

Reply:
I agree that Arg Ex is better, but this is breaking change? (Although I'd consider throwing an MsalEx a bug...)